packages/react-google-maps-api/src/useLoadScript.tsx

```diff
@@ -9,14 +9,25 @@ import { makeLoadScriptUrl, LoadScriptUrlOptions } from './utils/make-load-scrip
 
 import { defaultLoadScriptProps } from './LoadScript'
 import useIsMounted from './utils/useIsMounted'
+import { createUseAtMostOnce } from './utils/useAtMostOnce'
+import usePrevious from './utils/usePrevious'
 
 export interface UseLoadScriptOptions extends LoadScriptUrlOptions {
-  id?: string;
-  preventGoogleFontsLoading?: boolean;
+  id?: string
+  preventGoogleFontsLoading?: boolean
 }
 
 let previouslyLoadedUrl: string
 
+const useLoadScriptAtMostOnce = createUseAtMostOnce("useLoadScript should only be used at most once. The google script is loaded globally, and one hook's load could override another hook's load")
```

Comment:
I certainly don't like this. We shouldn't be forcing that upon the user. There is no technical reason why you cannot use it more times. If anything, it can show warning, but it certainly cannot throw an Error.

Reply:
if you look, it's actually creating a global variable, it's not using useRef() but creating that global varible hook-less.

But ok if you don't like I'll remove.

So you are ok to let user do something wrong (ie using twice this hook in the same page), and not warning him? (actually GoogleMaps will likely warn him anyway, unless we have a cleanup system to `delete google.maps` on hook unmount)

Reply:
I think it's quite a bad pattern. Hooks are generally for reusability purposes, people know it. It shouldn't be hook if it's not intended to be reused.

Cleanup is still in the "old" LoadScript and most likely was a reason for weird behavior and that's why I made this hook. There is no need. Google will replace `window.google` no matter if anything else is there.

Reply:
We could say exactly the same about components, yet there are components that are not meant to be mounted more than once per app and will throw if you try. For example react-navigation's appContainer.

Google will replace window.google, but with an error everytime, which will popup in my Sentry, which I don't like ;)

Reply:
well, there is an easy way to get rid of an error) 

remove `version` prop from `window.google` object

<img width="352" alt="Screenshot 2019-07-19 14 29 11" src="https://user-images.githubusercontent.com/2255993/61532190-b6e39b80-aa31-11e9-8a78-ab69c146aa89.png">

Reply:
Thanks, that looks like an useful tip to know.

What about if we:
- use a constant id
- when script is going to be removed/recreated due to url change, delete previous google.version to automatically avoid the error? That would be safer than deleting google.maps

Reply:
I have read the code of google maps, and even it is minified, the error string is searchable, and  error is shown only if `window.google.maps.version` is not undefined